Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/HPMotionController.cs

```diff
@@ -0,0 +1,271 @@
+                        interactionMapping.BoolData = !Mathf.Approximately(gripData, 0.0f);
+                        break;
+                    default:
+                        return;
+                }
+
+                // If our value changed raise it.
+                if (interactionMapping.Changed)
+                {
+                    // Raise bool input system event if it's available
+                    if (interactionMapping.BoolData)
+                    {
+                        CoreServices.InputSystem?.RaiseOnInputDown(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                    }
+                    else
+                    {
+                        CoreServices.InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                    }
+                }
+
+                switch (interactionMapping.InputType)
+                {
+                    case DeviceInputType.Trigger:
+                        var triggerData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Trigger);
+                        interactionMapping.FloatData = triggerData;
+                        break;
+                    case DeviceInputType.Grip:
+                        var gripData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Grasp);
+                        interactionMapping.FloatData = gripData;
+                        break;
+                    default:
+                        return;
+                }
+
+                // If our value changed raise it.
+                if (interactionMapping.Changed)
+                {
+                    // Raise float input system event if it's enabled
+                    CoreServices.InputSystem?.RaiseFloatInputChanged(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, interactionMapping.FloatData);
+                }
+            }
+        }
+
+        private static readonly ProfilerMarker UpdateDualAxisDataPerfMarker = new ProfilerMarker("[MRTK] HPController.UpdateDualAxisData");
+
+        /// <summary>
+        /// Update the touchpad / thumbstick input from the device
+        /// </summary>
+        protected virtual void UpdateDualAxisData(MixedRealityInteractionMapping interactionMapping, MotionControllerState controllerState)
+        {
+            using (UpdateDualAxisDataPerfMarker.Auto())
+            {
+                Debug.Assert(interactionMapping.AxisType == AxisType.DualAxis);
+
+                InputFeatureUsage<Vector2> axisUsage;
+
+                // Update the interaction data source
+                switch (interactionMapping.InputType)
+                {
+                    case DeviceInputType.ThumbStick:
+                        axisUsage = CommonUsages.primary2DAxis;
+                        break;
+                    case DeviceInputType.Touchpad:
```

Comment:
The HP controller doesn't have a touchpad, and I also don't see where these `axisUsage`s are used below. Left over copy from the XRSDK controller?

Reply:
Even after removal though, we probably still want some `interactionMapping.InputType == DeviceInputType.ThumbStick` check somewhere.